sound/soc/sof/sof-audio.c

```diff
@@ -45,10 +45,11 @@ static int sof_kcontrol_setup(struct snd_sof_dev *sdev, struct snd_sof_control *
 
 static int sof_dai_config_setup(struct snd_sof_dev *sdev, struct snd_sof_dai *dai)
 {
+	struct sof_ipc_dai_config *config;
 	struct sof_ipc_reply reply;
-	struct sof_ipc_dai_config *config = dai->dai_config;
 	int ret;
 
+	config = &dai->dai_config[dai->current_config];
```

Comment:
Yikes, good catch. FYI @juimonen

Reply:
this should be a formal Fixes: at this point, the original commit was merged by Mark Brown if I am mistaken?

Reply:
hmm thats right. I'll leave this one here and submit a separate PR to fix the upstream code.

Reply:
Not following @ranj063. The fixup! mechanism is only useful if the base commit is still in topic/sof-dev-rebase. we will squash the result before sending upstream
If the ship has sailed and the base commit to be fixed is already in Mark's tree, then we can't use fixup! but only a bona fide patch with a Fixes tag.